weather/src/js/components/WeatherApp.jsx

```diff
@@ -0,0 +1,94 @@
+import React from 'react';
+import { Table,td,tr,th,thead,tbody } from 'react-bootstrap';
+import { getLocationCoords, getWeatherData } from '../api/weather-api';
+
+export default class WeatherApp extends React.Component {
+
+  constructor() {
+    super(); // A call to the class we are extending: React.Component
+
+    this.state = {
+      data : {
+         forecastList : [],
+         city: 'Loading...',
+         country: 'Loading...'
+      }
+    };
+
+    this.componentDidMount = this.componentDidMount.bind(this);
```

Comment:
you don't have to do this. `ComponentDidMount` will have the context of the current component